x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -827,7 +827,7 @@ func safelyStoreAgentInfo(s saver, reader io.Reader) error {
 	for i := 0; i <= maxRetriesstoreAgentInfo; i++ {
 		backExp.Wait()
 		err = storeAgentInfo(s, reader)
-		if err != filelock.ErrAppAlreadyRunning {
+		if !stderror.Is(err, filelock.ErrAppAlreadyRunning) {
```

Comment:
I originally forgot to invert this check which was causing the `elastic-agent container` bootstrap to fail.
It now works